src/tbp/monty/frameworks/environments/embodied_data.py

```diff
@@ -379,16 +379,16 @@ def add_distractor_objects(
 
     def reset_agent(self):
         logging.debug("resetting agent------")
-        self._observation, self.motor_system._policy.state = self.dataset.reset()
+        self._observation, state = self.dataset.reset()
         self._counter = 0
 
         # Make sure to also reset action variables when resetting agent during
         # pre-episode
         self._action = None
         self._amount = None
-        self.motor_system._policy.state[self.motor_system._policy.agent_id][
-            "motor_only_step"
-        ] = False
+        state[self.motor_system._policy.agent_id]["motor_only_step"] = False
```

Comment:
Since we access it quite a few times here, it might make sense to assign `self.motor_system._policy.agent_id` to a local variable, e.g. `agent_id` to make all these indexes into `state` shorter and easier to read. The lines are pretty noisy otherwise.

I guess it's actually spread across several methods here, so it might be a narrower change than I'm thinking. 🤷

Reply:
Yes, to this and many other things I didn't address here 🙂. I'm deferring holding on to `agent_id` and reusing it as future work. This PR aims to reduce the number of `_policy.state` interactions.